src/Token/FactoryTokenProvider.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/*
+ * This file is part of the Mercure Component project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Component\Mercure\Token;
+
+/**
+ * @author Saif Eddin Gmati <azjezz@protonmail.com>
+ *
+ * @experimental
+ */
+final class FactoryTokenProvider implements TokenProviderInterface
+{
+    private $factory;
+    private $publish;
+    private $subscribe;
+
+    public function __construct(TokenFactoryInterface $factory, array $publish, array $subscribe)
+    {
+        $this->factory = $factory;
+        $this->publish = $publish;
+        $this->subscribe = $subscribe;
+    }
+
+    public function getToken(): string
```

Comment:
Using `jwt` instead of `token` would be more in sync with the terms we use on the Mercure website. (Just a thought, I've no strong opinion about this naming).

Reply:
done 👍